app/mock_svc.py

```diff
@@ -37,5 +48,22 @@ def __init__(self, services):
         sim_response = await self.data_svc.get('sim_response', dict(ability_id=ability['ability_id'], paw=paw))
         if not sim_response:
             return '', 0
+        if '|SPAWN|' in self.agent_svc.decode_bytes(sim_response[0]['response']):
+            if await self._spawn_new_sim(link):
+                return self.agent_svc.encode_string('spawned new agent'), sim_response[0]['status']
+            return self.agent_svc.encode_string('failed to spawn new agent'), 1
         return sim_response[0]['response'], sim_response[0]['status']
 
+    async def _spawn_new_sim(self, link):
+        filtered = [a for a in self.agents if not a['enabled']]
+        run_on = (await self.data_svc.get('core_agent', dict(paw=link['paw'])))[0]
+        command_actual = self.agent_svc.decode_bytes(link['command'])
+        target = None
+        for agent in filtered:
```

Comment:
can this block be more like:
```
    async def _spawn_new_sim(self, link):
        filtered = [a for a in self.agents if not a['enabled']]
        run_on = (await self.data_svc.get('core_agent', dict(paw=link['paw'])))[0]
        command_actual = self.agent_svc.decode_bytes(link['command'])
        for agent in filtered:
            box, user = agent['paw'].split('$')
            if user in command_actual and box in command_actual and run_on['platform'] == agent['os']:
                await self.start_agent(agent)
                return True
        return False
```

Reply:
Yeah, with the changes from before this becomes a lot cleaner.